src/ResolvedPeers.h

```diff
@@ -110,6 +116,35 @@ class ResolvedPeers: public RefCountable
     size_type availablePaths = 0;
 };
 
+/// Binds a candidate connection with its index in ResolvedPeers
+class ResolvedPeer
+{
+public:
+    explicit ResolvedPeer(const Comm::ConnectionPointer &conn, const ResolvedPeers::size_type pos) : connection_(conn), position_(pos) {}
+
+    ResolvedPeer() : connection_(nullptr), position_(ResolvedPeers::npos) {}
+
+    explicit operator bool() const { return static_cast<bool>(connection_); }
+
+    Comm::Connection *operator->() const { assert(connection_); return connection_.getRaw(); }
+
+    Comm::Connection &operator *() const { assert(connection_); return *connection_; }
+
+    /// resolved peer path expressed using a Comm::Connection
+    const Comm::ConnectionPointer &connection() const { return connection_; }
+
+    /// resolved peer path expressed using an (unconnected) Comm::Connection
+    const Comm::ConnectionPointer &address() const { return connection_; }
+
+    void connection(const Comm::ConnectionPointer &conn) { connection_ = conn; }
+
+private:
+    Comm::ConnectionPointer connection_; ///< (the address of) a path
```

Comment:
We need probably to add a comment, clarifying that 'connection_' may be other from the one initially obtained from 'destinations'.